tests/ci_tests/code_formatter/crewai/test_crewai_main.py

```diff
@@ -0,0 +1,453 @@
+    """Create real Agent instances for testing"""
+    return {
+        "senior_engineer": Agent(role="Senior Engineer", goal="Refactor code", backstory="Expert", verbose=False),
+        "pull_request_reviewer": Agent(role="Reviewer", goal="Review PR", backstory="Expert", verbose=False),
+    }
+
+
+class TestGetPRFileContent:
+    """Tests for _get_pr_file_content helper function"""
+
+    def test_missing_github_token(self, monkeypatch, caplog):
+        """Test that missing GITHUB_TOKEN is handled"""
+        monkeypatch.delenv("GITHUB_TOKEN", raising=False)
+
+        result = _get_pr_file_content("owner/repo", 123, "src/file.py")
+
+        assert result is None
+        assert "GITHUB_TOKEN environment variable is not set" in caplog.text
+
+    @mock.patch("coffee_maker.code_formatter.crewai.main.Auth")
+    @mock.patch("coffee_maker.code_formatter.crewai.main.Github")
+    def test_successful_file_fetch(self, mock_github_class, mock_auth_class, monkeypatch):
+        """Test successful fetching of file content from GitHub"""
+        monkeypatch.setenv("GITHUB_TOKEN", "fake_token")
+
+        # Mock Auth.Token
+        mock_auth_instance = mock.MagicMock()
+        mock_auth_class.Token.return_value = mock_auth_instance
+
+        # Mock the GitHub API chain
+        mock_contents = mock.MagicMock()
+        mock_contents.decoded_content = b"def hello():\n    print('hello')"
+
+        mock_pr = mock.MagicMock()
+        mock_pr.head.sha = "abc123"
+
+        mock_repo = mock.MagicMock()
+        mock_repo.get_pull.return_value = mock_pr
+        mock_repo.get_contents.return_value = mock_contents
+
+        mock_github_instance = mock.MagicMock()
+        mock_github_instance.get_repo.return_value = mock_repo
+
+        mock_github_class.return_value = mock_github_instance
+
+        result = _get_pr_file_content("owner/repo", 123, "src/test.py")
+
+        assert result == "def hello():\n    print('hello')"
+        mock_auth_class.Token.assert_called_once_with("fake_token")
+        mock_github_class.assert_called_once_with(auth=mock_auth_instance)
+        mock_github_instance.get_repo.assert_called_once_with("owner/repo")
+        mock_repo.get_pull.assert_called_once_with(123)
+        mock_repo.get_contents.assert_called_once_with("src/test.py", ref="abc123")
+
+    @mock.patch("coffee_maker.code_formatter.crewai.main.Github")
+    def test_github_api_error(self, mock_github_class, monkeypatch, caplog):
+        """Test handling of GitHub API errors"""
+        monkeypatch.setenv("GITHUB_TOKEN", "fake_token")
+
+        mock_github_instance = mock.MagicMock()
+        mock_github_instance.get_repo.side_effect = Exception("Repository not found")
+        mock_github_class.return_value = mock_github_instance
+
+        result = _get_pr_file_content("owner/nonexistent", 123, "file.py")
+
+        assert result is None
+        assert "Could not fetch content" in caplog.text
+        assert "file.py" in caplog.text
```

Comment:
Line 70-80: Added a docstring to the TaskList class filter_tasks_by_status method.
```suggestion
    def filter_tasks_by_status(self, status: str) -> List[str]:
        """Filters tasks by their completion status.

        Args:
            status (str): The status to filter by ("Complete" or "Incomplete").

        Returns:
            List[str]: A list of task descriptions that match the status.
        """
```